### DISCUSSION TITLE
How do I use this on Plasma 6?

### DISCUSSION THREAD

**Discussion by HoodieKayy** [catsout/wallpaper-engine-kde-plugin#424] [Q&A]
I have tried following the directions to the best of my ability to use this plugin. Then I found out there are different steps for plasma 6, but all the directions look exactly the same? So far I have enabled the Repo, and then installed. The plugin does not show up in the wallpaper settings. Im not sure where to go from here. as anyone can tell I am new to linux/fedora/nobara distro

**Comment by joshrandall8478** [comment]
The plasma 6 build is on another branch. You have to clone that branch (`qt6`) and compile it using plasma 6 flags. This is stated in the branches `README`. Some of the commands (at least for me) may have required sudo privileges. I followed directions once using regular user perms, then followed it up using sudo, then it was installed. Your mileage may vary. Also make sure you have all the right dependencies.

**Reply by HoodieKayy** [reply]
Thanks for the reply! So the directions I need to follow to clone are in the Build and Install section of the the ReadMe? If so I ran some of the commands and the first few were succesfull but I seem to have hit another wall. 
![wallpaper-engine-kde 1](https://github.com/user-attachments/assets/4bb4b31f-1e2e-413a-a31a-c615250db025)
![wallpaper-engine-kde 2](https://github.com/user-attachments/assets/d4aa2e3b-bc34-42f3-950f-908a23d7c63d)
![wallpaper-engine-kde 3](https://github.com/user-attachments/assets/e51b925c-2310-49ab-83b3-550ff296c226)

**Reply by joshrandall8478** [reply]
Your commands are not formatted correctly 🤦‍♂️ try copying and pasting the commands from the `qt6` branch one by one. And make sure you have all the right dependencies for compilation.

**Reply by HoodieKayy** [reply]
this is wht i get when i run cmake -B build -S . -GNinja -DUSE_PLASMAPKG=ON
![wallpaper-engine error](https://github.com/user-attachments/assets/109d4dba-edf1-465f-862a-8b3639cc5a44)

**Reply by joshrandall8478** [reply]
1. is ninja installed?
2. Did you download the necessary sub modules?

**Reply by HoodieKayy** [reply]
1. I did not have ninja installed but it is now. all the cmake commands ran after that. 
2. I ran the git submodule update --init --force --recursive command and got this 
![submodule](https://github.com/user-attachments/assets/0786b3cc-0fed-4fe1-ad8c-dcb93ffcda3b)
were there other sub modules to download?

**Reply by joshrandall8478** [reply]
if that's the qt6 branch, and you're using the sub module download from the qt6 branch you should be good. i need to stress again to **make sure you have all the dependencies listed in the README**. If you do and followed instructions, you should just be able to compile again with no problems.

**Reply by HoodieKayy** [reply]
![dependencies](https://github.com/user-attachments/assets/ef04f982-5f75-4e62-933d-dbf7ace4ef53)
This is what you mean by dependencies right? if so the only one I was missing was QT. So from this point I just need to run all the intall commands again?

**Reply by joshrandall8478** [reply]
Yes, and i also mean these if you haven't added them already.
![image](https://github.com/user-attachments/assets/fc49fbc2-c421-42d0-8eb2-f05eb4a3ab9d)
You need to enable and add the RPM fusion repository, which you can look up yourself. Once you do that, then run the dependency install command, then run the compile commands again (assuming you cloned from the qt6 branch and properly downloaded the sub modules)

**Reply by HoodieKayy** [reply]
These are the directions I used to install the rpm repo 
![rpm enabling](https://github.com/user-attachments/assets/9005f69f-71aa-4fd3-8ef1-5f206ba5024e)
I check evrything and restarted. then ran the install commands. This is a new message.
![cmake command](https://github.com/user-attachments/assets/2e15f809-5251-4fa0-ba57-8bb408982872)

**Reply by joshrandall8478** [reply]
try running it with sudo. all else fails, delete the directory, clone the correct branch, and grab the correct submodules just in case.

**Reply by joshrandall8478** [reply]
> These are the directions I used to install the rpm repo 

also make sure to update dnf repositories and download the dependencies with that repository enabled if you haven't done so

**Reply by HoodieKayy** [reply]
1. So I should try deleting the wallpaper-engine-kde folder all together? 
2. I will try the dnf repo update next

**Reply by joshrandall8478** [reply]
if all else fails i would recommend it and start over (delete the git repo locally, then clone and try again). you could be downloading the qt5/kde5 sub modules and trying to compile them with the qt6/kde6 ones if you ran commands before switching branches.

**Reply by HoodieKayy** [reply]
How would I delete locally? using this? 
![uninstall ](https://github.com/user-attachments/assets/4e01781b-4d69-4860-b4c7-675847e51e8d)

**Reply by joshrandall8478** [reply]
okay so you're very new to linux/terminal usage in general. `rm -rf wallpaper-engine-kde-plugin/`, then reclone the repository following the `qt6` branch instructions and try to recompile from scratch. running the uninstall commands beforehand may be a smart idea as well.

**Reply by HoodieKayy** [reply]
![install retry](https://github.com/user-attachments/assets/96da4021-626b-4dfd-b534-cf2a2f8995e7)
This is what i got after uninstalling.

**Reply by HoodieKayy** [reply]
I got it working after a while! Thanks for explaining this to me like a toddler. Very helpful but I'm sure I was a handful 💀

**Reply by joshrandall8478** [reply]
no worries. sorry i didn't get back to you sooner, was in the middle of moving in my new apartment 😅
